### DISCUSSION TITLE
Using tailwind css in custom css?

### DISCUSSION THREAD

**Discussion by trevhunter** [jpanther/congo#77] [Q&A] [answered]
Is there a way to use tailwindcss at the site level alongside the tailwind in the theme? [In the docs](https://jpanther.github.io/congo/docs/advanced-customisation/#overriding-the-stylesheet) it describes using `custom.css` just in terms of plain old CSS. As I override some of the partials at the site level or create new layouts, I'd like to be able to use tailwind too. 

* Creating a new `main.css` at the site level gets me tailwind at the site level, but it clobbers the theme's `main.css`. 
* I haven't quite figured out how to get tailwind to use `custom.css` at the site level. It keeps looking for `main.css` when I try to build or serve the site. 

I'm using the go modules way of using the theme. I guess one way would be to manually copy the theme in so postcss/tailwind processes the site and theme together, but I'm hoping for a more elegant solution. 

P.S. Thanks for providing such a good theme with great attention to detail and docs.

**Comment by jpanther** [comment, marked as answer]
I actually require custom Tailwind for my own site that uses Congo and so the theme is designed with this in mind but as you say, you do need to include the theme in your project separately to go modules.

As long as you have your site set up with the theme in a subfolder, you can then simply issue this command at the site level and it will generate a new `main.css` file at the site level `assets/css/compiled/` directory:

```shell
./themes/congo/node_modules/tailwindcss/lib/cli.js -c ./themes/congo/tailwind.config.js -i ./themes/congo/assets/css/main.css -o ./assets/css/compiled/main.css --jit
```

The CSS generated using this command will include any Tailwind that is required for the theme, as well as any Tailwind you are using in your site level layouts or content files. As Hugo assets at the site level take precedence over the theme this new file will automatically override the theme's `main.css`.

If you really want to keep using modules, in theory you could use the path to the Hugo cache directory and issue a similar command using the modules version of the theme but I think that's probably a less elegant solution than this.

**Reply by trevhunter** [reply]
Thank you @jpanther - you gave me inspiration to try something different and it seems to have worked. Still using the go module approach for the theme, I took your use of the `tailwindcss` cli to apply it to `custom.css` at the site level.

* I have a file `assets/css/custom.css.tw` with the tailwind directives in it.
* I have tailwindcss installed at the site level.
* I use the tailwind cli to compile it to `assets/css/custom.css` using the following:
```
./node_modules/tailwindcss/lib/cli.js -c ./tailwind.config.js -i ./assets/css/custom.css.tw -o ./assets/css/custom.css --jit
```

Once that's done, the site seems to pick up `custom.css` as it normally would, but this time with all the tailwind goodness. 

Now all I need to do is figure out how to invoke that command as part of hugo build (and add some icing on the cake to invoke it when something changes for hot reload) so I don't need to remember to do it all the time.

**Reply by jpanther** [reply]
Interesting solution. My only concern with this approach would be the unnecessary duplication of all the standard Tailwind declarations that are already in the `main.css` in your `custom.css` file.

**Reply by trevhunter** [reply]
Yeah, that's a risk, but hopefully not something that causes too much of an issue.

Ideally, I'm not creating _that_ many new things, and where there's overlap, it should resolve correctly, even with duplication (something I'll need to test though).

**Reply by jpanther** [reply]
A more detailed explanation of this in case it helps...  https://github.com/jpanther/congo/discussions/85#discussioncomment-2010789

**Reply by trevhunter** [reply]
That's an awesomely detailed and helpful answer. Thanks @jpanther.
